core/src/main/scala/scalafix/rewrite/Xor2Either.scala

```diff
@@ -13,16 +14,90 @@ case object Xor2Either extends Rewrite {
     }
     object DType extends Desugared[Type]
     object DTerm extends Desugared[Term]
+
+    //If we start making many of these, would it be worth having case classes with some structure?
+    //Possibly if I could select the appropriate based on a match, I have not been able to do that yet.
+    case class ReplaceType(original: Type,
+                           replacement: Type,
+                           newString: String)
+    val xorT: ReplaceType =
+      ReplaceType(t"cats.data.XorT", t"scala.util.EitherT", "EitherT")
+    val xor: ReplaceType =
+      ReplaceType(t"cats.data.Xor", t"scala.util.Either", "Either")
+    val xorLeft: ReplaceType =
+      ReplaceType(t"cats.data.Xor.Left", t"scala.util.Left", "Left")
+    val xorRight: ReplaceType =
+      ReplaceType(t"cats.data.Xor.Right", t"scala.util.Either.Right", "Right")
+
+    //experimenting to see if I can create lists of types to try to match.  No luck so far.
+    val replacementTypes: List[ReplaceType] =
+      List(xorT, xor, xorLeft, xorRight)
+
+    val termMap: Map[String, String] = Map(
+      "cats.data.Xor.Right.apply" -> "Right",
+      "cats.data.Xor.Left.apply" -> "Left"
+    )
+
     // NOTE. This approach is super inefficient, since we run semantic.desugar on
     // every case for every node in the tree. Ideally, we first match on the
     // syntax structure we want and then run semantic.desugar.
-    ast.collect {
-      case t @ DType(t"cats.data.Xor") =>
-        Patch(t.tokens.head, t.tokens.last, s"Either")
+    val tpes: Seq[(Type, ReplaceType)] = ast.collect {
+      case t @ DType(t"cats.data.XorT") => t -> xorT
+      case t @ DType(t"cats.data.Xor") => t -> xor
+      case t @ DType(t"cats.data.Xor.Left") => t -> xorLeft
+      case t @ DType(t"cats.data.Xor.Right") => t -> xorRight
+    }
```

Comment:
Here I am playing around with the idea of collecting interesting nodes and information about them and generating patches later.  It doesn't add much in this case but I thought it could be useful if we attempted to do some sort of fusion later, for example the same tree might have multiple sets of appends performed on it, in which case we could merge them.